### PR TITLE
Add `dev/` for configs for use by Nomad developers

### PR DESCRIPTION
**Pull request by dadgar**
This PR adds a folder `dev/` which can be used to store helpful commands
and configs for use when developing Nomad itself.

### CODE REVIEW

File: dev/docker-clients/client.nomad [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        # command = "nomad"
+        args         = ["agent"]
+        network_mode = "host"
+        volumes      = ["local/config:/etc/nomad", "/var/run/docker.sock:/var/run/docker.sock", "/tmp:/tmp"]
+        privileged   = true
+      }
+
+      resources {
+        cpu    = 300 # 500 MHz
+        memory = 100 # 256MB
```
Comment (schmichael) [inline dev/docker-clients/client.nomad]: comments are wrong

File: dev/docker-clients/client.nomad [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        data = <<EOF
+log_level = "DEBUG"
+data_dir = "/tmp/client{{ env "NOMAD_ALLOC_INDEX" }}"
+name = "client-{{ env "NOMAD_ALLOC_INDEX" }}"
+client {
+	enabled = true
+	servers = ["127.0.0.1:4647"]
+	options {
+	  "driver.raw_exec.enable" = "1"
+    }
```
Comment (schmichael) [inline dev/docker-clients/client.nomad]: whitespace is off. I've been trying to switch to 2 space indents based on some docs feedback from seth

File: dev/docker-clients/client.nomad [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    task "agent" {
+      driver = "docker"
+
+      config {
+        image = "djenriquez/nomad:v0.6.0"
+
+        # command = "nomad"
+        args         = ["agent"]
+        network_mode = "host"
+        volumes      = ["local/config:/etc/nomad", "/var/run/docker.sock:/var/run/docker.sock", "/tmp:/tmp"]
```
Comment (schmichael) [inline dev/docker-clients/client.nomad]: why do you mount the host's /tmp?

File: dev/docker-clients/persistent.hcl [hunk 4]
```diff
@@ -0,0 +1,16 @@
+log_level = "DEBUG"
+data_dir = "/tmp/server"
```
Comment (schmichael) [inline dev/docker-clients/persistent.hcl]: mind prefixing with `nomad-` so it's `/tmp/nomad-server`? Sometimes I want to wipe all nomad state from a system and a prefix makes it far easier.

File: dev/docker-clients/persistent.hcl [hunk 5]
```diff
@@ -0,0 +1,16 @@
+log_level = "DEBUG"
```
Comment (schmichael) [inline dev/docker-clients/persistent.hcl]: I also like adding:

```hcl
enable_debug = true
```

to all of my dev configs

File: dev/docker-clients/client.nomad (hunk 3 above)
Comment (dadgar) [inline dev/docker-clients/client.nomad]: So that you can access the clients directories without using the fs command. Just easier debugging.

**Pull request merged**